zone/client_process.cpp

```diff
@@ -289,6 +289,27 @@ bool Client::Process() {
 			entity_list.ScanCloseMobs(close_mobs, this, IsMoving());
 		}
 
+        if (RuleB(Inventory, LazyLoadBank) && Connected()) {
```

Comment:
Few things

* Indentation doesn't match the rest of the if branch
* `Connected()` is redundant because we check for it up top
* We need a timer around this to rate limit the costs of hitting this branch. `GetClosestBanker` is costly especially when being called 30 times a second when the client is in a state where their bank is not loaded yet. My suggestion would be to clamp the timer to 1000 milliseconds
* `GetClosestBanker` is compared with `USE_NPC_RANGE2` which is 40,000 units which is effectively zone wide. This is about as effective as just loading the bank as soon as the client zones in as long as there is a banker in the zone. I made a simpler helper method that could be used in place of this very cheaply. https://github.com/EQEmu/Server/pull/4462. `GetClosestBanker` utilizes the entire NPC list, which can in some cases be 600+, so 600 * 30 times a second doing distance checks * how many clients can add up quickly in zones where no banker exists but has many NPCs. `IsCloseToBanker` helper I added usually works against a very small list of NPC's closest already (600 units) and we don't need to do additional distance calculations to determine that.

Reply:
Yes, absolutely. `GetClosestBanker` isn't appropriate here after looking at it further. I was looking at the existing bank validation logic and a slower fill rate that needed a few seconds to fully populate the bank. That said, I'll fix this according to these suggestions.